src/task-metadata/task-metadata.controller.ts

```diff
@@ -14,28 +14,44 @@ import { CreateMetaTaskDto } from './dto/create-metaTask.dto';
 import { GetTaskMetadaDto } from './dto/get-tasks-metadata.dto';
 import { TaskMetadata } from './entity/task-metadata.entity';
 import { TaskMetadataService } from './task-metadata.service';
-
+import {
+  ApiParam,
+  ApiBearerAuth,
+  ApiCreatedResponse,
+  ApiOkResponse,
+  ApiTags,
+  ApiBody,
+  ApiUnauthorizedResponse,
+} from '@nestjs/swagger';
+@ApiTags('Task-Metadata')
 @Controller('task-metadata')
 @UseGuards(AuthGuard())
+@ApiBearerAuth('access-token')
+@ApiUnauthorizedResponse({ description: 'Unauthorized' })
 export class TaskMetadataController {
   private logger = new Logger('TasksController');
   constructor(private tasksMetadataService: TaskMetadataService) {}
 
   @Get('/:id/details')
+  @ApiOkResponse({ description: 'Get Task details' })
+  @ApiParam({ name: 'id', description: 'Task ID', type: String })
   getDetailsTaskById(
-    @Param('id') id: string,
+    @Param('id') taskId: string,
     @Query() task: Task,
   ): Promise<TaskMetadata> {
-    return this.tasksMetadataService.getTaskDetail(task, id);
+    return this.tasksMetadataService.getTaskDetail(task, taskId);
```

Comment:
The same thing, task may disappear, we already have the task identifier as **taskId**.